posthog/api/decide.py

```diff
@@ -414,27 +416,62 @@ def get_feature_flags_response_or_body(
     _record_feature_flag_metrics(team, feature_flags, errors, data)
 
     # Format response based on API version
-    return _format_feature_flags_response(feature_flags, feature_flag_payloads, errors, api_version)
+    return _format_feature_flags_response(feature_flags, feature_flag_payloads, flags_details, errors, api_version)
 
 
 def _format_feature_flags_response(
-    feature_flags: dict[str, Any], feature_flag_payloads: dict[str, Any], errors: bool, api_version: int
+    feature_flags: dict[str, Any],
+    feature_flag_payloads: dict[str, Any],
+    flags_details: Optional[dict[str, Any]],
+    errors: bool,
+    api_version: int,
 ) -> dict[str, Any]:
     """Format feature flags response according to API version."""
     active_flags = {key: value for key, value in feature_flags.items() if value}
 
     if api_version == 2:
         return {"featureFlags": active_flags}
-    elif api_version >= 3:
+    elif api_version == 3:
         return {
             "featureFlags": feature_flags,
             "errorsWhileComputingFlags": errors,
             "featureFlagPayloads": feature_flag_payloads,
         }
+    elif api_version >= 4:
+        return {
+            "flags": _format_feature_flag_details(flags_details),
+            "errorsWhileComputingFlags": errors,
+        }
     else:  # v1
         return {"featureFlags": list(active_flags.keys())}
 
 
+def _format_feature_flag_details(flags_details: Optional[dict]) -> dict:
+    if flags_details is None:
+        return {}
+
+    flag_details = {}
+    for flag_key, flag_value in flags_details.items():
+        flag_details[flag_key] = {
+            "key": flag_key,
+            "enabled": flag_value.match.match,
+            "variant": flag_value.match.variant,
+            "reason": {
+                "code": flag_value.match.reason.value,
+                "condition_index": flag_value.match.condition_index,
+                "description": None,  # TBD add description.
```

Comment:
open TBD? What would you ideally want to put here?

Reply:
I implemented this now.